contracts/DistributionCreator.sol

```diff
@@ -650,98 +443,179 @@ contract DistributionCreator is UUPSHelper, ReentrancyGuardUpgradeable {
+        // if the campaign doesn't last at least one second
+        if (newCampaign.duration == 0) revert CampaignDurationIsZero();
+        // if the reward token is not whitelisted as an incentive token
+        if (rewardTokenMinAmount == 0) revert CampaignRewardTokenNotWhitelisted();
+        // if the amount distributed is too small with respect to what is allowed
+        if ((newCampaign.amount * HOUR) / newCampaign.duration < rewardTokenMinAmount) revert CampaignRewardTooLow();
+
+        if (newCampaign.creator == address(0)) newCampaign.creator = msg.sender;
+
+        // Computing fees: these are waived for whitelisted addresses and if there is a whitelisted token in a pool
+        uint256 _fees = campaignSpecificFees[newCampaign.campaignType];
+        if (_fees == 0) _fees = defaultFees;
+        uint256 campaignAmountMinusFees = _computeFees(_fees, newCampaign.amount, newCampaign.rewardToken);
+        newCampaign.amount = campaignAmountMinusFees;
+
+        newCampaign.campaignId = campaignId(newCampaign);
+
+        if (_campaignLookup[newCampaign.campaignId] != 0) revert CampaignAlreadyExists();
+        _campaignLookup[newCampaign.campaignId] = campaignList.length + 1;
+        campaignList.push(newCampaign);
+        emit NewCampaign(newCampaign);
+
+        return newCampaign.campaignId;
+    }
+
+    /// @notice Creates a distribution from a deprecated distribution type
+    function _createDistribution(DistributionParameters memory newDistribution) internal returns (uint256) {
+        _createCampaign(_convertDistribution(newDistribution));
+        // Not gas efficient but deprecated
+        return campaignList[campaignList.length - 1].amount;
+    }
+
+    /// @notice Converts the deprecated distribution type into a campaign
+    function _convertDistribution(
+        DistributionParameters memory distributionToConvert
+    ) internal view returns (CampaignParameters memory) {
+        uint256 wrapperLength = distributionToConvert.wrapperTypes.length;
+        address[] memory whitelist = new address[](wrapperLength);
+        address[] memory blacklist = new address[](wrapperLength);
+        uint256 whitelistLength;
+        uint256 blacklistLength;
+        for (uint256 k = 0; k < wrapperLength; k++) {
+            if (distributionToConvert.wrapperTypes[k] == 0) {
+                whitelist[whitelistLength] = (distributionToConvert.positionWrappers[k]);
+                whitelistLength += 1;
+            }
+            if (distributionToConvert.wrapperTypes[k] == 3) {
+                blacklist[blacklistLength] = (distributionToConvert.positionWrappers[k]);
+                blacklistLength += 1;
+            }
+        }
+
+        assembly {
+            mstore(whitelist, whitelistLength)
+            mstore(blacklist, blacklistLength)
+        }
+
+        return
+            CampaignParameters({
+                campaignId: distributionToConvert.rewardId,
```

Comment:
In general the rewardId was not required to be computed before the creation. So if someone recreate a distribution from this contract it may create a wrong one

Reply:
Shouldn't we recompute it?

Reply:
Here it works well because it's recomputed when creating a distribution

Reply:
It is not recomputed here `_createDistribution`